app/controllers/api/v2/hosts_controller.rb

```diff
@@ -262,6 +262,28 @@ def power
         end
       end
 
+      api :GET, '/hosts/:id/power_status', N_('Fetch the status of whether the vm is powered on or not')
+      param :id, :identifier_dottable, required: true
+
+      def power_status
+        vm = @host.compute_resource.find_vm_by_uuid(@host.uuid)
+
+        unless @host.supports_power? && vm
```

Comment:
Power operations are supported also on bare-metal via BMC, why there is the `vm` clause?

Reply:
I see the method is designed to only work with VMs. We do have `api :PUT, "/hosts/:id/power", N_("Run a power operation on host")` for BMC operations, does it make sense to implement just GET for the same method and make it working with both BMC and VMs?

Reply:
Sounds good